airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -82,31 +113,105 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
             return True, error_msg
         return True, None
     
-    def streams(self, config: Mapping[str, Any]) -> Iterable[Stream]:
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         """
         Replace the streams below with your own streams.
 
         :param config: A Mapping of the user input configuration as defined in the connector spec.
         """
+        project_id = config["project_id"]
         self._auth = BigqueryAuth(config)
         streams = config.get("streams", [])
         sync_method = config["replication_method"]["method"]
-        streams_catalog = []
-        
-        for dataset in BigqueryDatasets(project_id=config["project_id"], authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
+        slice_range = float(config.get("slice_range", 525600)) # 1 year by default
+        fallback_start =  datetime.strptime("0001-01-01T00:00:00.000Z", '%Y-%m-%dT%H:%M:%S.%f%z')
+        change_history_start = datetime.now(tz=pytz.timezone("UTC")) - timedelta(days=7) + timedelta(seconds=30)
+        normal_streams = []
+        concurrent_streams = []
+
+        for stream in streams:
+                dataset_id, table_id = stream['parent_stream'].split(".")
+                where_clause = stream["where_clause"]
+                stream_name = stream["name"]
+                if sync_method == "Standard":
+                    table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, stream_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
+                else:
+                    try:
+                        table_obj = TableChangeHistory(project_id, dataset_id, table_id, stream_name, where_clause=where_clause, fallback_start=change_history_start, slice_range=slice_range, authenticator=self._auth)
+                        next(table_obj.read_records(sync_mode=SyncMode.full_refresh))
+                    except exceptions.HTTPError as error:
+                        self.logger.warning(f"Table had no inserts in the last seven days so no change history is available causing the error: {str(error.response.text)}")
+                        table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, stream_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
```

Comment:
I don't think we can switch from `History` to `Standard` like that as the format of the state would be different. Is there something I'm missing?

Reply:
I will not do this anymore following your suggestion of only returning the streams that support `Change history` in one of the previous comments, here's the update https://github.com/artefactory/airbyte/pull/28/commits/1891c26877e43fdaea5e73419272b3d7dff17303